### PR TITLE
OCaml: improve reading function and add functions not in 4.02

### LINKED ISSUE #15: OCaml minimum version may be too recent
The OCaml minimum version is 4.06. The problem is it not packaged in any Debian based distributions. I saw for instance that there will be a Prologin event with a 4.05 OCaml version.

Should I try to make the OCaml generator compatible with old OCaml versions.
The new function I use is `List.init`. A possibility would be to use procedural OCaml and `Array.init` instead. Or I could create the lists with an other way.

What do you all think?

### PR DESCRIPTION
**Pull request by FardaleM**
Fixes #15

### CODE REVIEW

File: iorgen/parser_ocaml.py [hunk 1]
```diff
@@ -69,28 +69,27 @@ def read_line(type_: Type, input_data: Input) -> str:
...
-        return 'Scanf.scanf "%s@\\n" (fun x -> x)'
+        return 'read_line ()'
     if type_.main == TypeEnum.LIST:
         assert type_.encapsulated is not None
         if type_.encapsulated.main == TypeEnum.INT:
-            return ('Scanf.scanf "%s@\\n" (fun x -> {}List.map int_of_string '
-                    '(String.split_on_char \' \' x))'
-                    ).format('if String.equal "" x then [] else ' if type_.
-                             can_be_empty else "")
+            return ('read_line () |> fun x -> if x = "" then [] else String.split_on_char \' \' x'
```
Comment (Shaac) [inline iorgen/parser_ocaml.py]: You do not like the idea of removing `if x = "" then [] else ` if it is not necessary to specify it?

File: iorgen/parser_ocaml.py [hunk 2]
```diff
@@ -187,15 +186,45 @@ def method(input_data: Input, reprint: bool) -> List[str]:
...
+        main += INDENTATION + "let {} = {} in\n".format(
             var_name(var.name), read_lines(var.type, input_data))
     args = (var_name(i.name) for i in input_data.input)
-    output += "{}{} {}\n".format(INDENTATION, var_name(input_data.name),
-                                 " ".join(args))
+    main += "{}{} {}\n".format(INDENTATION, var_name(input_data.name),
+                               " ".join(args))
+    if "List.init " in main:
+        # This is a quick fix to avoid dependency of OCaml 4.06 for List.init
+        output += "module List = struct\n"
```
Comment (Shaac) [inline iorgen/parser_ocaml.py]: You choose to put this on the very top of the generated file. I guess it is generally a good practice to do so. But here, if possible, I think it is better to keep it below the function to complete. This is just some boilerplate, and should not be the very first think to be read.

File: iorgen/parser_ocaml.py (hunk 1 above)
Comment (FardaleM) [reply iorgen/parser_ocaml.py]: For input reading I prefer to be more careful than necessary. And it simplifies the code of iorgen a little bit.

File: iorgen/parser_ocaml.py (hunk 2 above)
Comment (FardaleM) [reply iorgen/parser_ocaml.py]: There is already type definition at the top. With type at the top and this module hack at the bottom it become hard to find the function to complete but if you think it is better to do so I can move this bellow the function.

Comment (Shaac) [reply iorgen/parser_ocaml.py]: OK

**Pull request merged**